temporalio/worker/workflow_instance.py

```diff
@@ -0,0 +1,1314 @@
+
+
+class UnsandboxedWorkflowRunner(WorkflowRunner):
+    """Workflow runner that does not do any sandboxing."""
+
+    async def create_instance(self, det: WorkflowInstanceDetails) -> WorkflowInstance:
+        """Create an unsandboxed workflow instance."""
+        # We ignore MyPy failing to instantiate this because it's not _really_
+        # abstract at runtime. All of the asyncio.AbstractEventLoop calls that
+        # we are not implementing are not abstract, they just throw not-impl'd
+        # errors.
+        return _WorkflowInstanceImpl(det)  # type: ignore[abstract]
+
+
+_T = TypeVar("_T")
+_Context: TypeAlias = Dict[str, Any]
+_ExceptionHandler: TypeAlias = Callable[[asyncio.AbstractEventLoop, _Context], Any]
+
+
+class _WorkflowInstanceImpl(
+    WorkflowInstance, temporalio.workflow._Runtime, asyncio.AbstractEventLoop
+):
+    def __init__(self, det: WorkflowInstanceDetails) -> None:
+        # No init for AbstractEventLoop
+        WorkflowInstance.__init__(self)
+        temporalio.workflow._Runtime.__init__(self)
+        self._payload_converter = det.payload_converter_class()
+        self._defn = det.defn
+        self._info = det.info
+        self._pending_commands: List[
+            temporalio.bridge.proto.workflow_commands.WorkflowCommand
+        ] = []
+        self._primary_task: Optional[asyncio.Task[None]] = None
+        self._time = 0.0
+        # Handles which are ready to run on the next event loop iteration
+        self._ready: Deque[asyncio.Handle] = collections.deque()
+        self._conditions: List[Tuple[Callable[[], bool], asyncio.Future]] = []
+        # Keyed by seq
+        self._pending_timers: Dict[int, _TimerHandle] = {}
+        self._pending_activities: Dict[int, _ActivityHandle] = {}
+        self._pending_child_workflows: Dict[int, _ChildWorkflowHandle] = {}
+        # Keyed by type
+        self._curr_seqs: Dict[str, int] = {}
+        # TODO(cretz): Any concerns about not sharing this? Maybe the types I
+        # need to lookup should be done at definition time?
+        self._type_lookup = temporalio.converter._FunctionTypeLookup(
+            det.type_hint_eval_str
+        )
+        self._exception_handler: Optional[_ExceptionHandler] = None
+        # The actual instance, instantiated on first _run_once
+        self._object: Any = None
+
+        # We maintain signals and queries on this class since handlers can be
+        # added during workflow execution
+        self._signals = dict(self._defn.signals)
+        self._queries = dict(self._defn.queries)
+
+        # Maintain buffered signals for later-added dynamic handlers
+        self._buffered_signals: Dict[
+            str, List[temporalio.bridge.proto.workflow_activation.SignalWorkflow]
+        ] = {}
+
+        # Create interceptors. We do this with our runtime on the loop just in
+        # case they want to access info() during init().
+        temporalio.workflow._Runtime.set_on_loop(asyncio.get_running_loop(), self)
```

Comment:
I think it may be better if create and init the interceptors lazily in `_run_once` where I instantiate the workflow class instead of here in the constructor